src/main/scala/dotty/dokka/ScalaContentBuilder.scala

```diff
@@ -0,0 +1,504 @@
+        buildBlock: ScalaPageContentBuilder#ScalaDocumentableContentBuilder => ScalaPageContentBuilder#ScalaDocumentableContentBuilder
+    ): ContentGroup = buildBlock(
+            ScalaDocumentableContentBuilder(dris, sourceSets, kind, styles, extra)
+        ).buildContent()
+
+    def contentForDocumentable(
+        d: Documentable,
+        kind: Kind = ContentKind.Main,
+        styles: Set[Style] = Set(),
+        extra: PropertyContainer[ContentNode] = PropertyContainer.Companion.empty(),
+        buildBlock: ScalaPageContentBuilder#ScalaDocumentableContentBuilder => ScalaPageContentBuilder#ScalaDocumentableContentBuilder
+    ): ContentGroup = {
+        buildBlock(
+            ScalaDocumentableContentBuilder(Set(d.getDri), d.getSourceSets.asScala.toSet, kind, styles, extra)
+        ).buildContent()
+    }
+
+    case class ScalaTableBuilder(
+        val mainDRI: Set[DRI],
+        val mainSourcesetData: Set[DokkaConfiguration$DokkaSourceSet],
+        val mainKind: Kind,
+        val mainStyles: Set[Style],
+        val mainExtra: PropertyContainer[ContentNode],
+        val cells: List[ContentGroup] = List()
+    ) {
+        private def addChild(c: ContentGroup) = copy(cells = cells :+ c)
+
+        def cell(
+            dri: Set[DRI] = mainDRI,
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet] = mainSourcesetData,
+            kind: Kind = mainKind,
+            styles: Set[Style] = mainStyles,
+            extra: PropertyContainer[ContentNode] = mainExtra)(
+            buildBlock: Function1[ScalaPageContentBuilder#ScalaDocumentableContentBuilder, ScalaPageContentBuilder#ScalaDocumentableContentBuilder]
+        ): ScalaTableBuilder = addChild(self.contentForDRIs(dri, sourceSets, kind, styles, extra, buildBlock))
+
+        def build() = cells
+    }
+
+    case class ScalaDivergentBuilder(
+        val groupID: ContentDivergentGroup.GroupID,
+        val mainDRI: Set[DRI],
+        val mainKind: Kind,
+        val mainStyles: Set[Style],
+        val mainExtra: PropertyContainer[ContentNode],
+        val implicitlySourceSetHinted: Boolean,
+        val instances: List[ContentDivergentInstance] = List()
+    ) {
+        private def addChild(c: ContentDivergentInstance) = copy(instances = instances :+ c)
+
+        def buildContent() = ContentDivergentGroup(
+            instances.asJava,
+            DCI(mainDRI.asJava, mainKind),
+            mainStyles.asJava,
+            mainExtra,
+            groupID,
+            implicitlySourceSetHinted
+        )
+
+        def instance(
+            dri: Set[DRI],
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet],
+            kind: Kind = mainKind,
+            styles: Set[Style] = mainStyles,
+            extra: PropertyContainer[ContentNode] = mainExtra
+        )(
+            buildBlock: Function1[ScalaPageContentBuilder#ScalaDivergentInstanceBuilder, ScalaPageContentBuilder#ScalaDivergentInstanceBuilder]
```

Comment:
Write `X => T` instead of `Function1[X, Y]`?

Reply:
I realized that when I had half of ContentBuilder rewritten ;)